blockchain/chain/src/store/chain_store.rs

```diff
@@ -177,8 +189,7 @@ where
     }
 
     /// Returns the currently tracked heaviest tipset.
-    pub fn heaviest_tipset(&self) -> Option<Arc<Tipset>> {
-        // TODO: Figure out how to remove optional and return something every time.
+    pub fn heaviest_tipset(&self) -> Arc<Tipset> {
```

Comment:
API design-wise, IMHO it's better to use `pub fn heaviest_tipset_ref(&self) -> &Arc<Tipset>` so that it's cloned only when the caller really needs owned data

Reply:
Cloning an `Arc` is cheap and we want to let go of the mutex as soon as possible.